bosh-setup/azuredeploy.json

```diff
@@ -2,149 +2,114 @@
-      "metadata": {
-        "description": "name of the subnet for CloudFoundy"
-      }
-    },
     "vmName": {
       "type": "string",
       "metadata": {
         "description": "name of the vm, will be used as DNS Name for the Public IP used to access the Virtual Machine"
       }
     },
-    "vmSize": {
-      "type": "string",
-      "defaultValue": "Standard_D1",
-      "allowedValues": [
-        "Standard_A1",
-        "Standard_A2",
-        "Standard_A3",
-        "Standard_A4",
-        "Standard_D1",
-        "Standard_D2",
-        "Standard_D3",
-        "Standard_D4"
-      ],
-      "metadata": {
-        "description": "Size of vm"
-      }
-    },
     "adminUsername": {
       "type": "string",
       "metadata": {
-        "description": "User name for the Virtual Machine."
-      }
-    },
-    "adminPassword": {
-      "type": "securestring",
-      "metadata": {
-        "description": "Password for the Virtual Machine."
+        "description": "User name for the Virtual Machine"
       }
     },
-    "enableDNSOnDevbox": {
-      "type": "bool",
-      "defaultValue": true,
-      "allowedValues": [
-        true,
-        false
-      ],
+    "sshKeyData": {
+      "type": "string",
       "metadata": {
-        "description": "A default DNS will be setup in the devbox if it is true."
+        "description": "Please copy the content of your SSH RSA public key and paste it here. You can use \"ssh-keygen -t rsa -b 2048\" to generate your SSH key pairs."
       }
     },
     "tenantID": {
       "type": "string",
-      "defaultValue": "TENANT-ID",
+      "minLength": 36,
+      "maxLength": 36,
       "metadata": {
         "description": "ID of the tenant. See https://github.com/cloudfoundry-incubator/bosh-azure-cpi-release/blob/master/docs/guidance.md"
       }
     },
     "clientID": {
       "type": "string",
-      "defaultValue": "CLIENT-ID",
+      "minLength": 36,
+      "maxLength": 36,
       "metadata": {
         "description": "ID of the client. See https://github.com/cloudfoundry-incubator/bosh-azure-cpi-release/blob/master/docs/guidance.md"
       }
     },
     "clientSecret": {
       "type": "securestring",
-      "defaultValue": "CLIENT-SECRET",
       "metadata": {
         "description": "secret of the client. See https://github.com/cloudfoundry-incubator/bosh-azure-cpi-release/blob/master/docs/guidance.md"
       }
+    },
+    "autoDeployBosh": {
+      "type": "string",
+      "defaultValue": "disabled",
+      "allowedValues": [
+        "enabled",
+        "disabled"
+      ],
+      "metadata": {
+        "description": "The flag allowing to deploy Bosh automatically or not"
+      }
     }
   },
   "variables": {
-    "api-version": "2015-05-01-preview",
+    "apiVersion": "2015-06-15",
     "location": "[resourceGroup().location]",
-    "storageAccountType": "Standard_GRS",
+
+    "defaultStorageAccountName": "[concat('cfdefaultsa', uniqueString(resourceGroup().id))]",
```

Comment:
Having prefix before the uniqueString just makes it more clear for users.
You mean all storage accounts which start with the same prefix may be clustered together, right?
If so, I will follow your suggestion to have uniqueString before the prefix.